spec/models/events/pull_request_spec.rb

```diff
@@ -76,45 +96,70 @@
     it { is_expected.to have_many(:events) }
   end
 
-  context 'jobs' do
-    describe '#opened' do
-      it 'enqueues opened' do
+  describe 'actions' do
+    subject { create :pull_request, payload: raw_payload }
+
+    describe '#review_request' do
+      it 'creates a review request' do
         expect {
-          described_class.resolve(action: 'opened')
-        }.to have_enqueued_job(PullRequestJobs::Opened)
+          subject.review_requested
+        }.to change(ReviewRequest, :count).by(1).and change(User, :count).by(2)
       end
     end
 
     describe '#closed' do
-      it 'enqueues closed' do
+      it 'sets status closed' do
         expect {
-          described_class.resolve(action: 'closed')
-        }.to have_enqueued_job(PullRequestJobs::Closed)
+          subject.closed
+        }.to change { subject.reload.closed? }.from(false).to(true)
       end
     end
 
-    describe '#review_requested' do
-      it 'enqueues review requested' do
+    describe '#open' do
+      it 'sets status open' do
+        subject.closed!
+
         expect {
-          described_class.resolve(action: 'review_requested')
-        }.to have_enqueued_job(PullRequestJobs::ReviewRequested)
+          subject.open
+        }.to change { subject.reload.open? }.from(false).to(true)
       end
     end
 
-    describe '#review_request_removed' do
-      it 'enqueues review removal' do
+    describe '#merged' do
+      it 'sets status merged' do
         expect {
-          described_class.resolve(action: 'review_request_removed')
-        }.to have_enqueued_job(PullRequestJobs::ReviewRequestRemoved)
+          subject.merged
+        }.to change { subject.reload.merged? }.from(false).to(true)
       end
     end
 
-    describe '#merged' do
-      it 'enqueues merged' do
+    describe '#review_request_removed' do
+      let!(:pull_request) { create :pull_request_with_review_requests, payload: raw_payload }
+
+      it 'sets status to removed' do
+        review_request = User.find_by!(github_id: raw_payload[:requested_reviewer][:id])
```

Comment:
Can't you get the `review_request` from the `pull_request`? Isn't that easier?

Reply:
At this point `pull_request` has 5 `review_requests` and i have to find the `review_request` which has the `reviewer` with `github_id` `raw_payload[:requested_reviewer][:id]`